commit 5a924f86bc278189e2e2369687a6baa412b10837
Author: Michael Still <mikal@stillhq.com>
Date:   Mon Aug 3 10:22:06 2026 +1000

    Install the kerbside-proxy wheel into kerbside images.
    
    Kerbside recently split out the proxying element which shovels SPICE
    packets around from the python business logic and re-implemented
    the high bandwidth proxy component in rust. The proxy is still pip
    installable because of a maturin wrapper, and the kerbside package
    depends on the kerbside-proxy package correctly.
    
    However, it might be desirable for developers or deployers to be
    able to optionally override the kerbside-proxy wheel when building
    images. This patch therefore exposes such an override mechanism.
    
    Change-Id: Iaa556368a1d995a6764bd8b47833cc9472b1bb5d
    Signed-off-by: Michael Still <mikal@stillhq.com>

diff --git a/kolla/docker/kerbside/kerbside-base/Dockerfile.j2 b/kolla/docker/kerbside/kerbside-base/Dockerfile.j2
index 77f881399..32b92d67d 100644
--- a/kolla/docker/kerbside/kerbside-base/Dockerfile.j2
+++ b/kolla/docker/kerbside/kerbside-base/Dockerfile.j2
@@ -33,6 +33,11 @@ RUN ln -s kerbside-source/* kerbside \
            {{ macros.install_pip(kerbside_plugins_pip_packages) }}; \
        fi \
     && {{ macros.install_pip(kerbside_pip_packages | customizable("pip_packages"), false) }} \
+    && if ls /kerbside/proxy-wheels/*.whl >/dev/null 2>&1; then \
+           {{ macros.install_pip(['/kerbside/proxy-wheels/*.whl'], false) }}; \
+       else \
+           {{ macros.install_pip(['kerbside-proxy'], false) }}; \
+       fi \
     && rm -f /etc/machine-id
 
 {% block kerbside_base_footer %}{% endblock %}
